### PR TITLE
clean up core & 1x pom files

### PR DESCRIPTION
**Pull request by igorbernstein2**
- disable DuplicateClass checks from provided scope
- remove transitive deps
- remove mapreduce byo-hadoop artifact because it shades hbase deps
- remove hbase deps from -tools and replace logback with reload4j
- make annotation jars provided
- switch from assembly plugin to shade plugin for tools because it filters out provided deps by default
- remove guava from -tools, its overkill

Change-Id: I23af0b0cf84239238e8e482278ed447ddc786d4f

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 1]
```diff
@@ -257,46 +227,12 @@ limitations under the License.
...
         </executions>
       </plugin>
-
-      <plugin>
-        <groupId>com.google.cloud.bigtable.test</groupId>
-        <artifactId>bigtable-build-helper</artifactId>
-        <version>2.7.3-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
-        <executions>
-          <execution>
-            <id>verify-provided</id>
```
Comment (kolea2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml:274]: how come we are removing these checks?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml [hunk 2]
```diff
@@ -110,64 +105,67 @@
...
-              <goal>enforce</goal>
+              <goal>shade</goal>
             </goals>
             <configuration>
-              <rules>
-                <bannedDependencies>
+              <createDependencyReducedPom>true</createDependencyReducedPom>
+              <filters>
+                <filter>
+                  <artifact>*:*</artifact>
```
Comment (kolea2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:135]: for my understanding, what's this doing?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml [hunk 3]
```diff
@@ -110,64 +105,67 @@
...
+                    <exclude>META-INF/DEPENDENCIES</exclude>
+                    <exclude>META-INF/versions/9/module-info.class</exclude>
                   </excludes>
-                </bannedDependencies>
-              </rules>
+                </filter>
+              </filters>
+              <transformers>
+                <transformer
+                    implementation="org.apache.maven.plugins.shade.resource.ApacheLicenseResourceTransformer"/>
```
Comment (kolea2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:148]: same q here, what's this doing?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml [hunk 4]
```diff
@@ -110,64 +105,67 @@
...
+                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+              </transformers>
             </configuration>
           </execution>
         </executions>
       </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
```
Comment (kolea2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:160]: would you be able to move this back to reduce the diff?

File: pom.xml [hunk 5]
```diff
@@ -150,6 +150,44 @@ limitations under the License.
...
+                    <version>[3.0,)</version>
+                  </requireMavenVersion>
+                  <requireJavaVersion>
+                    <version>[1.8,)</version>
+                  </requireJavaVersion>
+                  <requireUpperBoundDeps/>
+                  <banDuplicateClasses>
+                    <scopes>
+                      <scope>compile</scope>
+                      <!-- remove provided scope from google-cloud-shared -->
```
Comment (kolea2) [inline pom.xml:180]: can you reword this - took me a minute. Maybe: "override from google-cloud-shared-config to only check for compile deps"

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml (hunk 1 above)
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml:274]: We are not, just moving it up

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml (hunk 2 above)
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:147]: Its telling maven shade plugin that the exclude below apply to all deps

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml (hunk 3 above)
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml:160]: Its concatenating the licenses & service files

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml (hunk 4 above)
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/pom.xml]: doen

File: pom.xml (hunk 5 above)
Comment (igorbernstein2) [reply pom.xml]: done

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 6]
```diff
@@ -211,24 +211,37 @@ limitations under the License.
...
-              <archive>
-                <manifest>
-                  <mainClass>com.google.cloud.bigtable.mapreduce.Driver</mainClass>
-                </manifest>
-              </archive>
-              <descriptors>
-                <descriptor>assembly/shaded.xml</descriptor>
-              </descriptors>
+              <createDependencyReducedPom>true</createDependencyReducedPom>
+              <finalName>bigtable-hbase-1.x-tools-${project.version}-jar-with-dependencies</finalName>
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml]: nope :)

**Pull request merged**